### PR TITLE
[BEAM-3690] ban mockito-all and hamcrest-all

### PR DESCRIPTION
**Pull request by PandaMonkey**
See: https://issues.apache.org/jira/browse/BEAM-3690
ban mockito-all and hamcrest-all

### CODE REVIEW

**Comment by PandaMonkey** [comment]
The purpose of this change is to configure the maven enforcer plugin to ban hamcrest-all/mockito-all using https://maven.apache.org/enforcer/enforcer-rules/bannedDependencies.html.

File: sdks/java/nexmark/pom.xml [hunk 1]
```diff
@@ -118,6 +118,22 @@
...
+	     <groupId>org.apache.maven.plugins</groupId>
+	        <artifactId>maven-dependency-plugin</artifactId>
+	        <executions>
+	          <execution>
+	            <goals>
+	              <goal>analyze-only</goal>
+	            </goals>
+	            <configuration>
+	              <ignoredUnusedDeclaredDependencies>
+	                <ignoredUnusedDeclaredDependency>org.hamcrest:hamcrest-library:jar:${hamcrest.version}</ignoredUnusedDeclaredDependency>
```
Comment (lukecwik) [inline sdks/java/nexmark/pom.xml]: This seems to duplicate the lines below.

Are they needed?

Comment (lukecwik) [reply sdks/java/nexmark/pom.xml]: Can you fix the indentation?

**Comment by lukecwik** [comment]
Note that we don't use tabs for indentation. I am cleaning up/flattening this PR and merging.

**Pull request closed without merging**